### DISCUSSION TITLE
External links image caching

### DISCUSSION THREAD

**Discussion by davegoopot** [mozilla/hubs#3074] [General Questions & Discussion] [answered]
I'm working on a Hub created in Spoke with link elements. It seems that after the first time the link is rendered in a Hub the thumbnail image that is shown in the Hub is cached. If the linked page content changes, the Hub thumbnail does not update. 

As the owner of the Hub and can press "refresh" on the thumbnail in the Hub causing it to update, but when I then leave the Hub the link thumbnail reverts to the earlier cached image.

How can I get the Hub thumbnail to update and stay updated to the latest linked content please?

**Comment by johnshaughnessy** [comment, marked as answer]
> How can I get the Hub thumbnail to update and stay updated to the latest linked content please?

I believe you're correct in saying this is due to the image being cached on the backend. When you click "refresh", we update an internal "version" for that link that causes the cache to get busted, but only for that particular request.

One workaround for this may be add arbitrary query string parameters to the URL, but as far as I know there's no way to explicitly bust the cache that works across sessions. I can't quite tell if the image cache is purged after 5 minutes -- I'm not very familiar with this part of reticulum yet. 

https://github.com/mozilla/reticulum/blob/master/lib/ret/media_resolver.ex#L368

This may warrant a feature request in either the hubs repo or the reticulum repo.

**Reply by robinkwilson** [reply]
@davegoopot Please make an issue / feature request in the hubs repo https://github.com/mozilla/hubs/issues and link this discussion thread! Would help us immensely. Please link it here when finished!

**Reply by robinkwilson** [reply]
[Temporarily marked as answered] Please click "Unmark as answered" button, when you add more information.
